url054c08669f_jules2689_CSI4124_tgz-pJ8-SMThemeParkExperi3J8/src/simModel/RVPs.java

```diff
@@ -0,0 +1,243 @@
+	final static double MEAN_ARRL_SH_8 = 0.21;
+	final static double MEAN_ARRL_SH_9 = 0.19;
+	final static double MEAN_ARRL_SH_10 = 0.19;
+	final static double MEAN_ARRL_SH_11 = 0.17;
+	final static double MEAN_ARRL_SH_12 = 0.15;
+
+	final static double MEAN_ARRL_GI_1 = 0.18;
+	final static double MEAN_ARRL_GI_2 = 0.18;
+	final static double MEAN_ARRL_GI_3 = 0.23;
+	final static double MEAN_ARRL_GI_4 = 0.29;
+	final static double MEAN_ARRL_GI_5 = 0.25;
+	final static double MEAN_ARRL_GI_6 = 0.21;
+	final static double MEAN_ARRL_GI_7 = 0.21;
+	final static double MEAN_ARRL_GI_8 = 0.22;
+	final static double MEAN_ARRL_GI_9 = 0.20;
+	final static double MEAN_ARRL_GI_10 = 0.18;
+	final static double MEAN_ARRL_GI_11 = 0.15;
+	final static double MEAN_ARRL_GI_12 = 0.14;
+
+	final static double MEAN_ARRL_RC_1 = 0.16;
+	final static double MEAN_ARRL_RC_2 = 0.19;
+	final static double MEAN_ARRL_RC_3 = 0.21;
+	final static double MEAN_ARRL_RC_4 = 0.23;
+	final static double MEAN_ARRL_RC_5 = 0.21;
+	final static double MEAN_ARRL_RC_6 = 0.19;
+	final static double MEAN_ARRL_RC_7 = 0.20;
+	final static double MEAN_ARRL_RC_8 = 0.19;
+	final static double MEAN_ARRL_RC_9 = 0.21;
+	final static double MEAN_ARRL_RC_10 = 0.19;
+	final static double MEAN_ARRL_RC_11 = 0.17;
+	final static double MEAN_ARRL_RC_12 = 0.15;
+
+	// Constructor
+	public RVPs(SMThemePark model, Seeds sd) {
+		this.model = model;
+		// Set up distribution functions
+		interArrDistFP = new Exponential(1.0 / MEAN_ARRL_FP_1, new MersenneTwister(sd.arrFP));
+		interArrDistSH = new Exponential(1.0 / MEAN_ARRL_SH_1, new MersenneTwister(sd.arrSH));
+		interArrDistGI = new Exponential(1.0 / MEAN_ARRL_GI_1, new MersenneTwister(sd.arrGI));
+		interArrDistRC = new Exponential(1.0 / MEAN_ARRL_RC_1,new MersenneTwister(sd.arrRC));
+	}
+
+	//
+	protected double uCustomersFP() {
+		double mean = getMeanTimeFP();
+		double nxtArrival = model.getClock() + interArrDistFP.nextDouble(1.0 / mean);
+		if (nxtArrival > model.closingTime) {
+			nxtArrival = -1.0; // Ends time sequence
+		}
+		return (nxtArrival);
+	}
+
+	protected double uCustomersSH() {
+		double mean = getMeanTimeSH();
+		double nxtArrival = model.getClock() + interArrDistSH.nextDouble(1.0 / mean);
+		if (nxtArrival > model.closingTime) {
+			nxtArrival = -1.0; // Ends time sequence
+		}
+		return (nxtArrival);
+	}
+
+	protected double uCustomersGI() {
+		double mean = getMeanTimeGI();
+		double nxtArrival = model.getClock() + interArrDistGI.nextDouble(1.0 / mean);
+		if (nxtArrival > model.closingTime) {
+			nxtArrival = -1.0; // Ends time sequence
+		}
+		return (nxtArrival);
+	}
+
+	protected double uCustomersRC() {
+		double mean = getMeanTimeRC();
+		double nxtArrival = model.getClock() + interArrDistRC.nextDouble(1.0 / mean);
+		if (nxtArrival > model.closingTime) {
+			nxtArrival = -1.0; // Ends time sequence
+		}
+		return (nxtArrival);
+	}
+
+	//TODO: Add to Documentation
+	protected double getMeanTimeFP() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.